symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/message/MessageService.java

```diff
@@ -218,6 +248,56 @@ public V4Message send(@Nonnull String streamId, @Nonnull String message) {
     ));
   }
 
+  /**
+   * Sends a message to the stream ID passed in parameter.
+   *
+   * @param streamId    the ID of the stream to send the message to
+   * @param message     the message to send to the stream
+   * @return a {@link V4Message} object containing the details of the sent message
+   * @see <a href="https://developers.symphony.com/restapi/reference#create-message-v4">Create Message v4</a>
+   */
+  public V4Message send(@Nonnull String streamId, @Nonnull Message message) {
+    File tempFile;
+    Attachment attachment = message.getAttachment();
+    if (attachment != null) {
+      try {
+        String[] fileNameArrays = attachment.fileName().split("\\.");
```

Comment:
If I understand well, the split is only there to validate the string format of `attachment.fileName()`. If so, you could extract this logic to a `validateAttachmentFileName` method (or with similar name) and reuse `attachment.fileName()` when you create a tempFile.